tsengcoin-core/src/v1/block.rs

```diff
@@ -153,7 +153,40 @@ impl Block {
         return self.transactions;
     }
 
-    /// Like `to_txns` except it excludes the coinbase transaction.
+    pub fn get_merkle_root(&self) -> Option<Hash256> {
+        let mut txns: Vec<[u8; 32]> = self.transactions.iter().
+        map(|t| t.hash)
+        .collect();
+
+        // Adding a identity hash so the number of
+        // transactions is divisible by 2.
+        if txns.len() % 2 != 0 {
+            txns.push([0; 32]);
```

Comment:
Shouldn't this be the last hash duplicated? 
![alt text](https://www.oreilly.com/api/v2/epubs/9781491902639/files/images/msbt_0703.png)

This is from [Mastering Bitcoin](https://www.oreilly.com/library/view/mastering-bitcoin/9781491902639/ch07.html#merkle_trees), the example shows a Merkle tree from 3 hashes. The last hash is duplicated because there is an odd number of hashes

Reply:
Gotcha, I wasn't sure exactly how this worked so I just added something random but I'll make the change to copy the last hash.

Reply:
Also, I see you already have a merkle tree implementation in https://github.com/Dezzmeister/tsengcoin/blob/b9c7f7b24f492240ae72c57101ed614932e74385/tsengcoin-core/src/v1/block.rs#L590
so we can just go with yours.